### DISCUSSION TITLE
Alpine linux build in any format that works.

### DISCUSSION THREAD

**Discussion by FredsterCode** [rustdesk/rustdesk#3840] [General]
I'm trying to run Rustdesk client on Alpine linux for a small project. 
I tried  :
- To build with the deb files on Alpine. That was a long shot I'll admit. 
- The default docker build with everything as is, That did not build on Alpine (somewhat expected once I understood the debian need) 
- Building the default docker build on ubuntu and then tried to move it over to Alpine. That didn't work. (vague error of "file missing" when ran up Alpine)
- To update the base in the Docker file FROM value from debian to Alpine: latest. This also failed and I could not see a clean way to fix all the resulting broken dependencies. 
- The flatpak approach but I cant seem to replicate the needed flatpak build environment to get it to work. currently stopped at "Can't exec "autopoint": No such file or directory" possibly another version mismatch. I have installed the flatpak build and then upgraded and then also complied it from source so I should have the very latest!!  

Could someone with amazing skill please update one of the avenues to enable a build for Alpine linux. 
I know its not a premier league desktop os but its extra slim footprint makes is an interesting candidate for some situations.
I dont know which is easier to get working or I would have logged a bug or feature request for just that one build approach. 
My current thought is the Flatpak approach is the most versatile though its the one with the least write up in the documentation.  [thread here](https://github.com/rustdesk/rustdesk/discussions/1537)
Snap perhaps !! 
Around and around my thoughts go. Either way I'm out of my depth for solving this by myself. 

Thanks for reading and even thinking about it. 

EDIT:
I have also tried to build it on alpine linux with a borrowed Gist (originally for Rustdesk server on alpine)
After a lot of finding and installing missing packages ( its a really long list and I didn't make a list either) along with environment variables I'm at brick wall. My stopper is the scrap module/crate and specifically a file called gstreamer-video. It looks to be a feature for screen recording and such and used with Wayland. no wayland is not supported exactly but there are some workaround implemented. video recording and wayland are not a requirement for me. 
gstreamer-video is missing from the gstreamer packages on Alpine. I guess it may have been too complicated to add in for Alpine and Alpine is not typically thought of as a video streaming platform. 

If someone knows how to chop off scrap it might work. !

**Comment by ghost** [comment]
I want to second this, as it will enable it to work on Postmarket OS mobile devices as well.  Though the interface is going to be huge unless a device is connected to an external display, which I started a discussion on at #10495.
